extension-adinfo/src/adinfo.cpp

```diff
@@ -36,15 +51,45 @@ static const luaL_reg Module_methods[] =
 static void LuaInit(lua_State* L)
 {
     DM_LUA_STACK_CHECK(L, 0);
-    ADInfo_InitAdInfoExt();
     int top = lua_gettop(L);
     luaL_register(L, MODULE_NAME, Module_methods);
     lua_pop(L, 1);
 }
 
+static dmExtension::Result UpdateAdInfo(dmExtension::Params* params)
+{
+    if(g_ADInfoData.m_callback != 0 && g_ADInfoData.m_AdId != 0)
+    {
+        lua_State* L = dmScript::GetCallbackLuaContext(g_ADInfoData.m_callback);
+        DM_LUA_STACK_CHECK(L, 0);
+
+        if (dmScript::SetupCallback(g_ADInfoData.m_callback))
+        {
+            lua_newtable(L);
+            lua_pushliteral(L, "ad_ident");
+            lua_pushstring(L, g_ADInfoData.m_AdId);
+            lua_rawset(L, -3);
```

Comment:
Nitpick really, but this can be replaced with 2 lines. And I think it's a bit easier to remember:

```
        lua_pushstring(L, g_ADInfoData.m_AdId);
        lua_setfield(L, -2, "ad_ident");
```